WordPress/Sniffs/Functions/ReturnTypeSniff.php

```diff
@@ -0,0 +1,141 @@
+	 */
+	public function process_token( $stackPtr ) {
+		$colon = (int) $this->phpcsFile->findPrevious( T_COLON, $stackPtr - 1 );
+
+		// Space before colon disallowed.
+		if ( T_CLOSE_PARENTHESIS !== $this->tokens[ $colon - 1 ]['code'] ) {
+			$error = 'There must be no space before colon before a return type.';
+			$fix   = $this->phpcsFile->addFixableError( $error, $colon - 1, 'SpaceBeforeColon' );
+
+			if ( true === $fix ) {
+				$this->phpcsFile->fixer->beginChangeset();
+				$token = $colon - 1;
+				do {
+					$this->phpcsFile->fixer->replaceToken( $token, '' );
+					-- $token;
+				} while ( T_CLOSE_PARENTHESIS !== $this->tokens[ $token ]['code'] );
+				$this->phpcsFile->fixer->endChangeset();
+			}
+		}
+
+		// Only one space after colon.
+		if ( T_WHITESPACE !== $this->tokens[ $colon + 1 ]['code'] ) {
+			$error = 'There must be a space after colon and before return type declaration.';
+			$fix   = $this->phpcsFile->addFixableError( $error, $colon, 'NoSpaceAfterColon' );
+			if ( true === $fix ) {
+				$this->phpcsFile->fixer->addContent( $colon, ' ' );
+			}
+		} elseif ( ' ' !== $this->tokens[ $colon + 1 ]['content'] ) {
+			$error = 'There must be exactly one space after colon and before return type declaration.';
+			$fix   = $this->phpcsFile->addFixableError( $error, $colon + 1, 'TooManySpacesAfterColon' );
+			if ( true === $fix ) {
+				$this->phpcsFile->fixer->replaceToken( $colon + 1, ' ' );
+			}
+		}
+
+		$nullable = (int) $this->phpcsFile->findNext( T_NULLABLE, $colon + 1, $stackPtr );
+		if ( $nullable ) {
+			// Check if there is space after nullable operator.
+			if ( T_WHITESPACE === $this->tokens[ $nullable + 1 ]['code'] ) {
+				$error = 'Space is not not allowed after nullable operator.';
+				$fix   = $this->phpcsFile->addFixableError( $error, $nullable + 1, 'SpaceAfterNullable' );
+				if ( $fix ) {
+					$this->phpcsFile->fixer->replaceToken( $nullable + 1, '' );
+				}
+			}
+		}
+
+		$first   = $this->phpcsFile->findNext( Tokens::$emptyTokens, ( $nullable ?: $colon ) + 1, null, true );
+		$end     = $this->phpcsFile->findNext( array( T_SEMICOLON, T_OPEN_CURLY_BRACKET ), $stackPtr + 1 );
+		$last    = $this->phpcsFile->findPrevious( Tokens::$emptyTokens, $end - 1, null, true );
+		$invalid = (int) $this->phpcsFile->findNext( array( T_STRING, T_NS_SEPARATOR, T_RETURN_TYPE ), $first, $last + 1, true );
+		if ( $invalid ) {
+			$error = 'Return type declaration contains invalid token %s';
+			$data  = array( $this->tokens[ $invalid ]['type'] );
+			$fix   = $this->phpcsFile->addFixableError( $error, $invalid, 'InvalidToken', $data );
+			if ( true === $fix ) {
+				$this->phpcsFile->fixer->replaceToken( $invalid, '' );
+			}
+
+			return;
+		}
+
+		$returnType = trim( $this->phpcsFile->getTokensAsString( $first, $last - $first + 1 ) );
+
+		if ( $first === $last
+			&& ! in_array( $returnType, $this->simple_return_types, true )
+			&& in_array( strtolower( $returnType ), $this->simple_return_types, true )
```

Comment:
There is no upstream as such, so flipped array in 7bcea29eddf726e218c53d6a244e88f86fbf28e1.